development/mob_programming.md

```diff
@@ -0,0 +1,45 @@
+# Mob Programming for Research
+
+## "Mob programming"?
```

Comment:
Perhaps we could add a why here?  There is good information here, but even by the end I wasn't sure why or when I would use this.

Reply:
I probably put this too far down the document and it could be the headline - the "why" is the below, and the "when" is "whenever it's not completely obvious what the next steps are, and sometimes even then"

This isn't intended as a sales pitch though, more rules of thumb for getting the most out of the practise, you'll gain much more by trying it (using the rules of thumb to minimise the amount of patience you need to space) than reading about trying it :D 

_Remember that you are optimising for learning rather than delivery in the early stages; as a [mob programming coach](https://code.joejag.com/2018/how-i-learned-mobbing.html) in Glasgow put it, lifting from an African proverb, *If you want to go fast, go alone; if you want to go far, go together*

 Mob programming, especially at the start of a team or project, can feel slow; but there are a few ways in which it's inwardly speeding up - all working on the same code at once means there's no slow cycle of waiting for peer review or context switching to offer it, working branches can merge straight to `main` at the end of a session. Progress might feel like it's in smaller increments but it's also more continuously headed in a sound direction - avoid blind alleys, avoid thrashing on small blockages, quickly draw on multiple perspectives to detect fruitless overthinking. _